singer_sdk/streams/rest.py

```diff
@@ -92,7 +93,13 @@ def __init__(
         if path:
             self.path = path
         self._http_headers: dict = {}
-        self._requests_session = requests.Session()
+
+        self.connector = HTTPConnector(self.config)
+
+        # Override the connector's auth with the stream's auth
+        self.connector._auth = self.authenticator
```

Comment:
Is there a way to make this `auth` override part of the connector constructor, rather than manipulating the private attr? Alternatively maybe a helper method to the connector class ?

 ```suggestion
        self.connector.override_auth(self.authenticator)
```

A public `.authenticator` attr could also work (wrapping the existing `_auth` private attr), but is maybe less self-documenting 🤔